sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -717,9 +717,12 @@ case class JsonToStructs(
   override def withTimeZone(timeZoneId: String): TimeZoneAwareExpression =
     copy(timeZoneId = Option(timeZoneId))
 
+  private val variantAllowDuplicateKeys = SQLConf.get.getConf(SQLConf.VARIANT_ALLOW_DUPLICATE_KEYS)
```

Comment:
Actually, this is not proper approach of getting the SQL config. The expression can change its behaviour during analys, see https://github.com/apache/spark/pull/27658

Reply:
Thanks for pointing this out! It has been fixed now.